### DISCUSSION TITLE
To hookup third party OAuth servers to Serverpod

### DISCUSSION THREAD

**Discussion by braj065** [serverpod/serverpod#849] [Q&A] [answered]
Hi,

As a developer can i use third party OAuth servers like Keycloak with a cluster of Serverpod running instances, for authentication and authorization purpose?

Thanks.

**Comment by yahu1031** [comment, marked as answer]
@vikas-flutter Yes you can do it if there is an open package for that in dart. Else you can use their APIs and integrate them and have a callback.

**Reply by braj065** [reply]
@yahu1031 Can the following Keycloak library be used with Serverpod? Even though it is a middlware for Shelf.
https://pub.dev/packages/shelf_keycloak

**Reply by braj065** [reply]
@yahu1031 Let me know if the above one will work?

**Reply by yahu1031** [reply]
> @yahu1031 Let me know if the above one will work?

Never tried buddy, but you can actually give a shot and LMK if you got stuck somewhere.
